### DISCUSSION TITLE
How to store Swagger documentation in separate file(s)?

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#10333] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
description = """
ChimichangApp API helps you do awesome stuff. 🚀

## Items

You can **read items**.

## Users

You will be able to:

* **Create users** (_not implemented_).
* **Read users** (_not implemented_).
"""

tags_metadata = [
    {
        "name": "login",
        "description": "Manage items. So _fancy_ they have their own docs.",
        "externalDocs": {
            "description": "Items external docs",
            "url": "https://fastapi.tiangolo.com/",
        },
    },
]

app = FastAPI(
    title=" API",
    description=description,
    summary="Deadpool's favorite app. Nuff said.",
    version="0.0.1",
    terms_of_service="http://example.com/terms/",
    contact={
        "name": "Deadpoolio the Amazing",
        "url": "http://x-force.example.com/contact/",
        "email": "dp@x-force.example.com",
    },
    license_info={
        "name": "Apache 2.0",
        "url": "https://www.apache.org/licenses/LICENSE-2.0.html",
    },
    openapi_tags=tags_metadata
)
```

### Description

I have such working right now from documentation
And i want to transfer such documentation from .py code to separate files,like json,xml
I tried 
```
Swagger UI: served at /docs.
You can set its URL with the parameter docs_url.
ReDoc: served at /redoc.
You can set its URL with the parameter redoc_url
```.
And no success.

### Operating System

Linux

### Operating System Details

_No response_

### FastAPI Version

0.68.2

### Pydantic Version

1.10.12

### Python Version

3.10

### Additional Context

_No response_

**Comment by n8sty** [comment, marked as answer]
Are you looking to retrieve the `openapi.json` file that gets generated by FastAPI? If that's what you're looking for it should be available at /openapi.json. Otherwise, your question isn't clear to me.

**Comment by Ryandaydev** [comment]
I think it helps to consider that the fastapi is doing a couple of things here that are really nice:
- autogenerating the openapi.json file -- this is useful for importing into tools, generate code, etc. 
- hosting two sets API documentation using other open source projects (swagger ui and redoc)

You mentioned you would like a json format. If that is the case, as @n8sty mentioned it is available at the /openapi.json. You can download it and load it up in a tool like postman or swagger editor (https://editor.swagger.io/) to see the structure there.

It also sounds like you'd like some HTML docs that you can save offline or possibly host separately from the app. It is typical for companies to host swagger documentation on their developer portal using the swagger_dist version. You can use this method with a saved version of the openapi.json, or even pointing to a hosted fastapi end point, again using the /openapi.json URL.
Here are swagger docs with instructions: https://swagger.io/docs/open-source-tools/swagger-ui/usage/installation/
